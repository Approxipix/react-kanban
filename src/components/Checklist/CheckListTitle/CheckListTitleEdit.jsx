import React, { Component } from "react";
import { connect } from "react-redux";
import { bindActionCreators } from 'redux';
import { editCheckListTitle } from '../../../redux/checkListReducer/actions';
import ClickOutside from "../../ClickOutside";
import { Input, } from '../../BaseComponent';

class CheckListTitleEdit extends Component {
  constructor(props) {
    super(props);
    this.state = {
      title: props.checkListTitle || '',
    };
  }

  handleKeyDown = (e) => {
    if (e.keyCode === 27) {
      this.props.toggleEditing()
    }
  };

  handleChange = (value) => {
    this.setState({
      title: value,
    });
  };


  handleSubmit = (e) => {
    e.preventDefault();
    const { title } = this.state;
    const { actions } = this.props;
    if (!title)  return;
    actions.editCheckListTitle({
      checkListId: this.props.checkListId,
      checkListTitle: title,
    });
    this.props.toggleEditing()
  };

  render = () => {
    const { title } = this.state;
    return (
      <ClickOutside toggleOpened={this.props.toggleEditing}>
        <form onSubmit={this.handleSubmit}>
          <Input
            type="text"
            padding=".3rem .5rem"
            margin="0"
            size="1.2rem"
            weight="600"
            value={title}
            placeholder="Enter list title..."
            onKeyDown={this.handleKeyDown}
            onChange={(e) => this.handleChange(e.target.value)}
            onBlur={this.handleSubmit}
            spellCheck={false}
            autoFocus
          />
        </form>
      </ClickOutside>
    );
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      editCheckListTitle: editCheckListTitle,
    }, dispatch)
  };
}

export default connect(null, mapDispatchToProps)(CheckListTitleEdit);
