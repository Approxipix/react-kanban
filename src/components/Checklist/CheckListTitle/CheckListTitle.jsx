import React, { Component } from 'react';
import CheckListTitleEdit from './CheckListTitleEdit'
import styled from 'styled-components'

const Title = styled.h3`
  padding: .3rem .5rem;
  font-size: 1.2rem;
  font-weight: 600;
`;

class CheckListTitle extends Component {
  constructor(props) {
    super(props);
    this.state = {
      editTitle: false,
    }
  }

  toggleEditTitle = () => {
    this.setState({
      editTitle: !this.state.editTitle
    })
  };

  render() {
    const { checkListId, checkListTitle} = this.props;
    const { editTitle } = this.state;
    return (
      !editTitle ? (
          <Title onClick={() => this.toggleEditTitle()}>
            {checkListTitle}
          </Title>
        ) : (
          <CheckListTitleEdit
            toggleEditing={this.toggleEditTitle}
            checkListTitle={checkListTitle}
            checkListId={checkListId}
          />
        )
    );
  }
}

export default CheckListTitle;
