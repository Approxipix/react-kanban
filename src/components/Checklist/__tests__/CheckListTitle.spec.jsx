import React from 'react';
import { shallow } from 'enzyme';
import toJson from "enzyme-to-json";
import CheckListTitle from '../CheckListTitle/CheckListTitle';

describe('<CheckListTitle>', () => {
  let state;
  let props;
  let component;

  beforeEach(() => {
    state = {
      editTitle: false,
    };
    props = {
      checkListId: 'CheckListID',
      checkListTitle: 'CheckList Title',
    };

    component = shallow(<CheckListTitle {...props} />);
    component.setState({...state})
  });

  it('should render title if editTitle is false', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render CheckListTitleEdit component if editTitle is true', () => {
    component.instance().toggleEditTitle();
    expect(toJson(component)).toMatchSnapshot();
  });

  it('checks if toggleEditTitle sets editTitle correctly', () => {
    component.instance().toggleEditTitle();
    expect(component.instance().state.editTitle).toEqual(true);

    component.instance().toggleEditTitle();
    expect(component.instance().state.editTitle).toEqual(false);
  });

  it("should call toggleEditTitle after click on title", () => {
    const spy = jest.spyOn(component.instance(), "toggleEditTitle");

    component.find('Title').simulate('click');

    expect(spy).toHaveBeenCalledTimes(1);
  });
});
