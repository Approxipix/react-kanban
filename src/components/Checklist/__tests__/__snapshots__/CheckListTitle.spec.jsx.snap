// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckListTitle> should render CheckListTitleEdit component if editTitle is true 1`] = `
<Connect(CheckListTitleEdit)
  checkListId="CheckListID"
  checkListTitle="CheckList Title"
  toggleEditTitle={[Function]}
/>
`;

exports[`<CheckListTitle> should render title if editTitle is false 1`] = `
<Title
  onClick={[Function]}
>
  CheckList Title
</Title>
`;
